bika/lims/exportimport/instruments/varian/vistapro/icp.py

```diff
@@ -0,0 +1,265 @@
+from bika.lims.exportimport.instruments.resultsimport import \
+    InstrumentCSVResultsFileParser, AnalysisResultsImporter
+from bika.lims import bikaMessageFactory as _
+import json
+import traceback
+
+logger = logging.getLogger(__name__)
+
+title = "Varian Vista-PRO ICP"
+
+
+class VistaPROICPParser(InstrumentCSVResultsFileParser):
+    """ Vista-PRO Parser
+    """
+    def __init__(self, csv):
+        InstrumentCSVResultsFileParser.__init__(self, csv)
+        self._end_header = False
+        self._resultsheader = []
+        self._numline = 0
+
+    def _parseline(self, line):
+        if self._end_header is False:
+            return self.parse_headerline(line)
+        else:
+            return self.parse_resultsline(line)
+
+    def parse_headerline(self, line):
+        """ Parses header lines
+        """
+        if self._end_header is True:
+            # Header already processed
+            return 0
+
+        splitted = [token.strip() for token in line.split(',')]
+        if splitted[0] == 'Solution Label':
+            self._resultsheader = splitted
+            self._end_header = True
+        return 0
+
+    def parse_resultsline(self, line):
+        """ CSV Parser
+        """
+
+        splitted = [token.strip() for token in line.split(',')]
+        if self._end_header:
+            resid = splitted[0]
+        rawdict = {'DefaultResult': 'Soln Conc'}
+        rawdict.update(dict(zip(self._resultsheader, splitted)))
+        date_string = "{Date} {Time}".format(**rawdict)
+        date_time = DateTime(date_string)
+        rawdict['DateTime'] = date_time
+        element = rawdict.get("Element", "").replace(" ", "").replace(".", "")
+
+        # Set DefaultResult to 0.0 if result is "0" or "--" or '' or 'ND'
+        result = rawdict[rawdict['DefaultResult']]
+        column_name = rawdict['DefaultResult']
+        result = self.zeroValueDefaultInstrumentResults(column_name,
+                                                        result,
+                                                        line)
+        rawdict[rawdict['DefaultResult']] = result
+        #
+
+        val = re.sub(r"\W", "", element)
+        self._addRawResult(resid, values={val: rawdict}, override=False)
+
+        self.log(
+            "End of file reached successfully: ${total_objects} objects, "
+            "${total_analyses} analyses, ${total_results} results",
+            mapping={"total_objects": self.getObjectsTotalCount(),
+                     "total_analyses": self.getAnalysesTotalCount(),
+                     "total_results": self.getResultsTotalCount()}
+        )
+
+    def zeroValueDefaultInstrumentResults(self, column_name, result, line):
+        result = str(result)
+        if result.startswith('--') or result == '' or result == 'ND':
+            return 0.0
+
+        try:
+            result = float(result)
+            if result < 0.0:
+                result = 0.0
+        except ValueError:
```

Comment:
You can take advantage of `api.is_floatable` and `api.to_float` here. Something like:
```python
if api.is_floatable(result):
    result = api.to_float(result)
    return result > 0.0 and result or 0.0

self.err("No valid number ${result} in column (${column_name})",
...
```